src/MongoDB.EntityFrameworkCore/Extensions/MongoPropertyExtensions.cs

```diff
@@ -587,10 +593,12 @@ internal static bool IsOwnedTypeKey(this IReadOnlyProperty property)
                && ownership.Properties.All(fkProperty => pk.Properties.Contains(fkProperty));
     }
 
-    internal static bool IsOwnedCollectionShadowKey(this IReadOnlyProperty property) =>
-        property.FindContainingPrimaryKey()
-            is { Properties.Count: > 1 } && !property.IsForeignKey()
-                                         && property.ClrType == typeof(int)
-                                         && (property.ValueGenerated & ValueGenerated.OnAdd) != 0
-                                         && property.GetElementName().Length == 0;
+    internal static bool IsOwnedTypeOrdinalKey(this IReadOnlyProperty property) =>
+        property.DeclaringType as IReadOnlyEntityType is { } entityType
+        && entityType.IsOwned()
+        && property.ClrType == typeof(int)
+        && (property.ValueGenerated & ValueGenerated.OnAdd) != 0
+        && property.GetElementName().Length == 0
+        && !property.IsForeignKey()
```

Comment:
Theoretically, an owned type PK could be configured as an FK for something else. Do we need to exclude FKs?

Reply:
We should probably talk about this as I have questions...